CMakeLists.txt

```diff
@@ -31,11 +31,13 @@ cmake_dependent_option(YAML_MSVC_SHARED_RT
   "MSVC: Build yaml-cpp with shared runtime libs (/MD)" ON
   "MSVC" OFF)
 
-set(yaml-cpp-type STATIC)
-set(yaml-cpp-label-postfix "static")
 if (YAML_BUILD_SHARED_LIBS)
   set(yaml-cpp-type SHARED)
   set(yaml-cpp-label-postfix "shared")
+else()
+  set(yaml-cpp-type STATIC)
+  set(yaml-cpp-label-postfix "static")
+  add_definitions(-DYAML_CPP_STATIC_DEFINE)
```

Comment:
It seems more natural to me to add a definition for the shared lib, which is the unusual case. (I don't feel *super* strongly, but since it looks like it doesn't matter one way or another, I thought I'd say something.)

Reply:
I am not a specialist at CMake, so this change is made by virtue of my little knowledge in this assembly system. I guess you can change this section as you see fit.

Reply:
OK, I guess it really doesn't matter much. I'll merge the code now and any improvements can be made after the fact.